core/arch/arm/include/kernel/linker.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2017, Linaro Limited
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef __KERNEL_LINKER_H
+#define __KERNEL_LINKER_H
+
+#include <initcall.h>
+#include <kernel/pseudo_ta.h>
+#include <kernel/dt.h>
+#include <mm/core_mmu.h>
+#include <types_ext.h>
+
+/*
+ * Symbols exported by the link script.
+ */
+
+
+/*
+ * These addresses will be the start or end of the exception binary search
+ * index table (.ARM.exidx section)
+ */
+extern const uint8_t __exidx_start[];
+extern const uint8_t __exidx_end[];
```

Comment:
Minor comment: this *could* be guarded by `#ifdef ARM32` (and same in `kernel.ld.S`), and a couple of other symbols here could also be guarded by `#ifdef CFG_something`. But is it worth it? (it would probably make the file less readable for no benefit).